astroquery/cadc/core.py

```diff
@@ -609,7 +609,7 @@ def exec_sync(self, query, maxrec=None, uploads=None, output_file=None,
             if isinstance(output_file, str):
                 fname = output_file
             elif hasattr(output_file, 'name'):
-                fname = output_file.name
+                fname = str(output_file)
```

Comment:
Second block is meant of file handlers. Maybe all is needed is for the first if to be `if isinstance(output_file, (str, Path)): name = str(output_file)`

Reply:
ok, I can do that, it makes sense. Though it would also make sense to add a test for the file handler case.